pkg/rhelv2/rpm/rpm.go

```diff
@@ -77,33 +84,31 @@ func AddToDependencyMap(filename string, fileData analyzer.FileData, execToDeps,
 	}
 }
 
-func getCPEsUsingEmbeddedContentSets(files analyzer.Files) ([]string, error) {
-	defer metrics.ObserveListFeaturesTime(pkgFmt, "cpes", time.Now())
-
-	// Get CPEs using embedded content-set files.
-	// The files are stored in /root/buildinfo/content_manifests/ and will need to
-	// be translated using mapping file provided by Red Hat's PST team.
-	contents := getContentManifestFileContents(files)
+// getContentManifestSets returns the list of content sets defined in the content
+// manifest, if found in the files. Otherwise, `nil` or empty slice is returned.
+func getContentManifestSets(files analyzer.Files) ([]string, error) {
+	var contents []byte
+	for _, prefix := range contentManifestDirs.AsSlice() {
+		for name, file := range files.GetFilesPrefix(prefix) {
+			if strings.HasSuffix(name, ".json") {
+				// Return the first one found, as we are currently assuming there is only one per
+				// layer/node.
+				contents = file.Contents
+				break
+			}
+		}
+		if contents != nil {
+			break
+		}
```

Comment:
So if we have multiple locations to search for - `len(contentManifestDirs.AsSlice()) > 1`  and we find a content in the first location `contentManifestDirs[0]`, why are we are breaking the search here? Shouldn't we check also the other ones - i.e., `contentManifestDirs[1]`, `contentManifestDirs[2]` and so on?

The current behavior would suggest that we are searching all location but we expect that the first hit is correct. Is it safe to assume that?

Reply:
> Shouldn't we check also the other ones - i.e., contentManifestDirs[1], contentManifestDirs[2] and so on?

We want to check all the `contentManifestDirs` but stop on the first file found. That is because we don't expect to find content sets in more than one directory.

Initially, this code would only search in one directory, the default directory for content sets in certified RHEL-based images. Now we are adding RHCOS, which uses a different location -- they have a limitation in their build that prevents them from writing to `/root` at compose time -- but we don't expect to see content sets in both locations.

> The current behavior would suggest that we are searching all locations but we expect that the first hit is correct. Is it safe to assume that?

For RHCOS node scanning, we are starting to debate that assumption.

Based on the [original discussion](https://github.com/coreos/coreos-assembler/issues/2540), it seems the image build would collect all repos used and populate [one single file](https://github.com/coreos/coreos-assembler/blob/main/src/cmdlib.sh#L497). But [there are some reports](https://github.com/stackrox/scanner/pull/1055#discussion_r1072829203) of multiple content sets found "in the wild". Which might be caused by RHCOS customizations? We are not sure, but we are planning to follow up. In any case, maybe we should not try to capture that scenario without knowing more about it.

What do you think, @vikin91?

Reply:
Okay this convinces me. I understand that the potential case of multiple content sets json files will be handled in a separate PR.

Reply:
I haven't seen it in RHCOS, but I have in RHEL-based images, which makes me think it could always be possible in RHCOS unless told otherwise. For example, in https://catalog.redhat.com/software/containers/redhat-openjdk-18/openjdk18-openshift/58ada5701fbe981673cd6b10?container-tabs=overview I see this:

```
[jboss@3ae40f064f3c ~]$ ls /root/buildinfo/content_manifests/
redhat-openjdk-18-openjdk18-openshift-container-1.15-2.json
rhel-server-container-7.9-848.1669829396.json
```

Reply:
> we are planning to follow up

Do we have a ticket to track this?

Reply:
> we are planning to follow up

https://issues.redhat.com/browse/ROX-13794